src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngineResolver.cs

```diff
@@ -11,12 +14,18 @@ namespace Microsoft.PowerFx
     /// <summary>
     /// <see cref="INameResolver"/> implementation for <see cref="RecalcEngine"/>.
     /// </summary>
-    internal class RecalcEngineResolver : SimpleResolver
+    internal class RecalcEngineResolver : SimpleResolver, INameResolver2
     {
         private readonly RecalcEngine _parent;
         private readonly PowerFxConfig _powerFxConfig;
 
-        public RecalcEngineResolver(RecalcEngine parent, PowerFxConfig powerFxConfig)
+        public IReadOnlyDictionary<string, NameLookupInfo> GlobalSymbols => new ReadOnlyDictionary<string, NameLookupInfo>(_parent.Formulas.Select(f =>
+        {
+            var description = $"{f.Key} variable";
+            return (f.Key, Value: new NameLookupInfo(BindKind.PowerFxResolvedObject, f.Value.Value.Type._type, DPath.Root, 0, f.Value, displayName: new DName(description)));
+        }).ToDictionary(kvp => kvp.Key, kvp => kvp.Value));
+
```

Comment:
The ReadOnlyDictionary isn't necessary - dictionary already impls IReadOnlyDictionary. 

Can you share with below (ie, proposed Create function)


  _parent.Formulas
     .ToDictionary(
          kvp => kvp.Key, 
          kvp => Create(kvp.key, kvp.value)
       );